### DISCUSSION TITLE
Private modules over SSH not supported on windows callers

### DISCUSSION THREAD

**Discussion by pjmagee** [dagger/dagger#12234] [Help]
### What is the issue?

private-modules don't work with the Dagger Windows binary release, because Windows does not recognise `
export SSH_AUTH_SOCK`. This is because on Windows, SSH is handled through a system wide OpenSSH pipe. `\.\pipe\openssh-ssh-agent`. 

I think both documentation needs to reflect this and a discussion around features supported or unsupported with the different dagger binaries is needed.

Right now, I have not been able to confirm a working solution right now with WSL. But that would also mean not using the dagger windows binary release. It would mean using dagger linux distro, in WSL and then also having to setup/configure SSH Keys in WSL. This seems more like a huge work-around and will also make the dagger windows binary unusable.

**Comment by grouville** [comment]
Hey @pjmagee,
Thanks for the issue. Looking into it, just need to have access to our windows computer 🙏 

EDIT: got access to a Windows computer, taking a look this week on how to have native support on Windows 🙏

**Comment by pjmagee** [comment]
> Hey @pjmagee, Thanks for the issue. Looking into it, just need to have access to our windows computer 🙏
> 
> EDIT: got access to a Windows computer, taking a look this week on how to have native support on Windows 🙏

Thank you! Hopefully it's not too crazy... but it is different to mac/linux when I was looking into it on the surface.

**Comment by pjmagee** [comment]
FYI, I had some real pains trying to get something working in WSL. I do not store my SSH keys on my devices, I am always using 1Password where I am able to. 1Password integration into WSL is using some special interop to execute ssh.exe from `Windows OS` but this does not use the `SSH_AUTH_SOCK`. My attempt was to try and install Linux distribution of 1Password into WSL, however this turned out to be really painful and did not work properly for a few reasons, some to do with GUI issues and wslg.

If anyone else finds a workaround with Windows SSH Pipe & WSL for Dagger, please let me know, otherwise for the mean-time I am looking into temporary alternatives to try out dagger private modules on MacOS

**Comment by grouville** [comment]
Hey Patrick,

Took the time to dig a bit:
1. Buildkit does support npipes, Dagger should be able to, no reason we wouldn't. The current primitives are not there yet though, on our end: https://github.com/moby/buildkit/pull/2127/files
2. I'll dig a way to have a workaround with the WSL, to see if current implementation could be workaround'ed in the meantime

Then, it's a prioritization on our end:
I have seen that https://github.com/git-ecosystem/git-credential-manager is supported on all platforms.

Users are eager for [PAT support through git credential manager](https://github.com/dagger/dagger/issues/7990), so I would prefer to focus on that first than implementing this npipe support, even though it totally makes sense for unification

**Comment by marcosnils** [comment]
> 2\. I'll dig a way to have a workaround with the WSL, to see if current implementation could be workaround'ed in the meantime

I'm thinking that if we make the `npipes` approach work, WSL could probably leverage that as well since IIUC once the `ssh-agent` is started on windows, seems like the `npipe` that it creates can be accessed from WSL2 in this path: `./pipe/openssh-ssh-agent`

**Comment by jedevc** [comment]
Just to update this issue, we still don't support `npipes` *yet*, but still no reason we couldn't!

However, we do now support HTTPS auth for private modules, so for getting private modules on windows, you should be able to use that. See https://docs.dagger.io/api/remote-repositories/#https-authentication
